### DISCUSSION TITLE
Bug when upgrading media library from v8 to v10 (MediaCollection json)

### DISCUSSION THREAD

**Discussion by BARNZ** [spatie/laravel-medialibrary#3203] [Q&A]
When using the Media model directly to manually filter and query media from the database and return results from route/controller, the following code worked fine in v8:

```php
Route::get('all-media', fn () => Media::all());
```

In v9 and v10 this is broken - it now always returns an empty array `[]`.

Looking into things, the following appears to be the culprit:
https://github.com/spatie/laravel-medialibrary/blob/2d185517adbd1bcb423373fcb80f40e808abcec9/src/MediaCollections/Models/Collections/MediaCollection.php#L57

`Media:all()` now returns a MediaCollection which has its own `jsonSerialize` which always returns an empty array for the above scenario. I dont quite understand the purpose of this custom MediaCollection, but I would expect Media collections to serialize the same as any other laravel model collection.

My workaround for now is on my custom media model add the following override to prevent `MediaCollection` being used:

```php
public function newCollection(array $models = [])
{
    #return new MediaCollection($models);
     return new \Illuminate\Database\Eloquent\Collection($models);
}
```

Perhaps a fix for this use-case can be added, or maybe something that needs adding to the upgrade guide?

Quite often there is a need to chain some additional filters off a media builder then return the filtered media directly from a controller as a json array - so I would have thought this would be quite a common scenario. From what I can tell I cant see anyone else having logged this issue.

**Comment by BARNZ** [comment]
Related pull request: https://github.com/spatie/laravel-medialibrary/pull/3210

**Comment by Gompje** [comment]
thanks for this. puzzled why the collection always returned empty in json while the items where there.. figured it would be something like this.. personally I share your opinion to follow the standard Laravel things. its just easier to work, especially on larger projects.

Seems like your pull request is still to be merged; sadly without a ETA
welp, workaround for now I guess ;-) 
// I was lucky in that way I just needed to re-create the special collection to a normal one in 1 place and I got what I wanted. but still.. feels hacky

big thanks @BARNZ !!
